### DISCUSSION TITLE
Can't add plugin to Aspire RabbitMQ host

### DISCUSSION THREAD

**Discussion by jmfenoll** [microsoft/aspire#6968] [Q&A] [answered]
I'm trying to add a plugin to RabbitMQ.

```
        var rabbitMQ = builder.AddRabbitMQ("rabbitMQ", rabbitUser, rabbitPassword)
            .WithImage("rabbitmq", "3.13-management")
            .WithEnvironment("RABBITMQ_PLUGINS_DIR", "/opt/rabbitmq/plugins:/myplugins")
            .WithEndpoint("management", e =>
            {
                e.Port = 15674;
                e.TargetPort = 15672;
            })
            .WithBindMount($"{curDir}/compose/volumes/rabbitmq/plugins", "/myplugins")
            .WithArgs("rabbitmq-plugins", "enable", "rabbitmq_delayed_message_exchange");  // This command exits container
 ```

After executing .WithArgs("rabbitmq-plugins".....,  , container restarts, Aspire detects it and mark container as Exit in desktop.

![image](https://github.com/user-attachments/assets/09a5ddd9-4e16-4ee1-bb01-07e5c4c47b3e)

![image](https://github.com/user-attachments/assets/b56ffc83-b464-4a23-83e2-797f51058c88)

However, if I comment ..WithArgs("rabbitmq-plugins" line and execute the command directly in docker terminal, it works.

![image](https://github.com/user-attachments/assets/6777e264-9c47-43f0-8bf8-88f46e31402e)

How can I automate with Aspire the plugin install?

The plugin that I try to install is [delayed message exchange](https://github.com/rabbitmq/rabbitmq-delayed-message-exchange)

Thanks

**Comment by davidfowl** [comment, marked as answer]
You changed the entry point to run `rabbitmq-plugins` instead of running rabbitmq, that's why it quits. What you did directly in docker was interactively run a command in the context of the container. This would be a development scenario only.

We don't have first class support for this yet in the app model. As an alternative, you BindMount a shell script and execute it as the entrypoint and whatever commands you need to as well as the rabbitmq entrypoint.

Alternatively, you can build a custom docker container:

https://stackoverflow.com/a/52819989/45091
